packages/react-core/src/components/Timestamp/__tests__/Timestamp.test.tsx

```diff
@@ -80,92 +71,85 @@ test('Renders with custom formatting when customFormat is passed in', () => {
 
   expect(screen.getByText('2 days ago')).toBeInTheDocument();
 });
 
 test('Renders with a custom suffix when displaySuffix is passed in', () => {
-  render(
-    <Timestamp datetime={new Date(2022, 0, 1).toISOString()} date={new Date(2022, 0, 1)} displaySuffix="US Eastern" />
-  );
+  render(<Timestamp date={new Date('1 Jan 2022 00:00:00 EST')} displaySuffix="US Eastern" />);
 
   expect(screen.getByText('1/1/2022, 12:00:00 AM US Eastern')).toBeInTheDocument();
 });
 
-test('Renders with 12 hour time by default for US locale', () => {
-  render(<Timestamp datetime={new Date(2022, 0, 1, 13, 0).toISOString()} date={new Date(2022, 0, 1, 13, 0)} />);
+test('Renders with 12 hour time when is12Hour is passed', () => {
+  render(<Timestamp date={new Date(2022, 0, 1, 13, 0)} is12Hour />);
 
   expect(screen.getByText('1/1/2022, 1:00:00 PM')).toBeInTheDocument();
 });
 
-test('Renders with 12 hour time when is12Hour is passed in for US locale', () => {
-  render(
-    <Timestamp datetime={new Date(2022, 0, 1, 13, 0).toISOString()} date={new Date(2022, 0, 1, 13, 0)} is12Hour />
-  );
+test('Renders with 24 hour time when is12Hour is set to false', () => {
+  render(<Timestamp date={new Date(2022, 0, 1, 13, 0)} is12Hour={false} />);
 
-  expect(screen.getByText('1/1/2022, 1:00:00 PM')).toBeInTheDocument();
+  expect(screen.getByText('1/1/2022, 13:00:00')).toBeInTheDocument();
 });
 
-test('Renders with 24 hour time when is12Hour is set to false for US locale', () => {
-  render(
-    <Timestamp
-      datetime={new Date(2022, 0, 1, 13, 0).toISOString()}
-      date={new Date(2022, 0, 1, 13, 0)}
-      is12Hour={false}
-    />
-  );
+test('Renders with locale passed in', () => {
+  render(<Timestamp date={new Date(2022, 1, 1)} locale="en-GB" />);
 
-  expect(screen.getByText('1/1/2022, 13:00:00')).toBeInTheDocument();
+  expect(screen.getByText('01/02/2022, 00:00:00')).toBeInTheDocument();
+});
+
+test('Renders with 12 hour time by default for US locale', () => {
+  render(<Timestamp date={new Date(2022, 1, 1, 13, 0)} locale="en-US" />);
+
+  expect(screen.getByText('2/1/2022, 1:00:00 PM')).toBeInTheDocument();
+});
+
+test('Renders with 24 hour time for US locale when is12Hour is false', () => {
+  render(<Timestamp date={new Date(2022, 1, 1, 13, 0)} locale="en-US" is12Hour={false} />);
+
+  expect(screen.getByText('2/1/2022, 13:00:00')).toBeInTheDocument();
+});
+
+test('Renders with 12 hour time for a 24 hour locale when is12Hour is passed', () => {
+  render(<Timestamp date={new Date(2022, 1, 1, 13, 0)} locale="en-GB" is12Hour />);
+
+  expect(screen.getByText('01/02/2022, 1:00:00 pm')).toBeInTheDocument();
+});
+
+test('Renders with default class names', () => {
+  render(<Timestamp date={new Date('1 Jan 2022 00:00:00 EST')} />);
+
+  expect(screen.getByText('1/1/2022, 12:00:00 AM')).toHaveClass('pf-c-timestamp__text');
+  expect(screen.getByText('1/1/2022, 12:00:00 AM').parentElement).toHaveClass('pf-c-timestamp');
```

Comment:
🤔 I've been doing one class assertion per test so far, but this approach could help make testing easier without *much* drawback. The only issue I can think of with this approach is that (because there are multiple points of failure for the test) seeing it fail provides *slightly* less value as it would require further investigation to see which class isn't being applied.

I would be curious to hear others' thoughts about these two approaches to testing for default class names.

Reply:
That's a fair point! At least locally when testing *only* the timestamp suite, it does provide that context of what class isn't being applied. Running all of the tests locally definitely doesn't provide that context, and whether it would when checks are run in the PR could go either way.

For sake of clarity I'd be good with separating this to two tests (with components that have a *lot* of classes it *might* cause a bit of bloat to test each class individually, but since it's only 2 classes here I don't think it's really an issue).